Cython/Compiler/Main.py

```diff
@@ -660,12 +702,27 @@ def compile(source, options = None, full_module_name = None, **kwds):
     checking is requested, a CompilationResult is returned, otherwise a
     CompilationResultSet is returned.
     """
+    from ..Build.Cache import Cache
+
     options = CompilationOptions(defaults = options, **kwds)
+
+    # cache is enabled when:
+    # * options.cache is True (the default path to the cache base dir is used)
+    # * options.cache is the explicit path to the cache base dir
+    # * annotations are not generated
+    if options.cache:
+        cache_path = None if options.cache is True else options.cache
+        cache = Cache(cache_path) if not (options.annotate or Options.annotate) else None
+    else:
+        if (options.annotate or Options.annotate) and options.verbose:
+            sys.stderr.write(f'Cache is ignored when annotations are enabled.\n')
+        cache = None
```

Comment:
I don't think the comment matches the implementation. The cache is not enabled automatically when annotations are not generated, but the comment seems to suggest that.

I think enabling the cache should require `options.cache` to be true (or a path), right?

```suggestion
    # cache is enabled when:
    # * options.cache is True (the default path to the cache base dir is used)
    # * options.cache is the explicit path to the cache base dir
    # unless annotations are generated
    cache = None
    if options.cache:
        if options.annotate or Options.annotate:
            if options.verbose:
                sys.stderr.write('Cache is ignored when annotations are enabled.\n')
        else:
            cache_path = None if options.cache is True else options.cache
            cache = Cache(cache_path)
```